### ISSUE TITLE
Tools -> AVD Manager disappears sometimes with Flutter projects

### ISSUE THREAD

**Issue by maryx** [flutter/flutter-intellij#2159]
## Steps to Reproduce
(I'm doing this on Linux right now but have seen the same behavior on MacOS)
1. With a fresh install of Android Studio 3+, with Flutter plugin and SDK configured, open an existing Flutter project. 
2. The emulator selector says `<no devices>`, so I select a device I had made previously. It results in this:
![screenshot from 2018-04-26 09-34-06](https://user-images.githubusercontent.com/3357543/39319273-7768c07e-4935-11e8-9671-5f075d58011d.png)

3. Try to open the AVD Manager to start the emulator from there. https://flutter.io/setup-linux/ points to Tools -> AVD Manager to view my devices. However, the AVD option doesn't show up. It shows up if I create a blank Android project, which is what I was doing to overcome this issue until I found this answer https://stackoverflow.com/a/47173929/8468400 (I'm not sure if my projects are missing something but this is an example project where the AVD Manager disappears: https://github.com/flutter/udacity-course/tree/master/course/04_navigation/solution_04_navigation.)
![screenshot from 2018-04-26 09-35-46](https://user-images.githubusercontent.com/3357543/39319192-39bd61e4-4935-11e8-9584-4a63b0af81e0.png)

3. This is fine - I can do ctrl + shift + a, and type "AVD Manager" to bring up the AVD Manager. I start my device. This works successfully. I build my app successfully.

4. Subsequent development on this emulator, including hot reloads and restarts, works without any problems.

The two issues are:
- The emulator panic error
- Tools -> AVD Manager option disappears

I'm not sure if this is Flutter-IntelliJ-specific, my project-specific, Android Studio-specific, or something else, so feel free to move this issue.

## Version info
```
$ flutter doctor -v
[✓] Flutter (Channel beta, v0.3.1, on Linux, locale en_US.UTF-8)
    • Flutter version 0.3.1 at /usr/local/google/home/maryxia/github-forks/flutter
    • Framework revision 12bbaba9ae (6 days ago), 2018-04-19 23:36:15 -0700
    • Engine revision 09d05a3891
    • Dart version 2.0.0-dev.48.0.flutter-fe606f890b

[✓] Android toolchain - develop for Android devices (Android SDK 27.0.3)
    • Android SDK at /usr/local/google/home/maryxia/Android/Sdk
    • Android NDK location not configured (optional; useful for native profiling support)
    • Platform android-27, build-tools 27.0.3
    • Java binary at: /usr/local/google/home/maryxia/Downloads/android-studio/jre/bin/java
    • Java version OpenJDK Runtime Environment (build 1.8.0_152-release-1024-b01)
    • All Android licenses accepted.

[✓] Android Studio (version 3.1)
    • Android Studio at /usr/local/google/home/maryxia/Downloads/android-studio
    • Flutter plugin version 23.2.2
    • Dart plugin version 173.4700
    • Java version OpenJDK Runtime Environment (build 1.8.0_152-release-1024-b01)

[✓] VS Code (version 1.22.2)
    • VS Code at /usr/share/code
    • Dart Code extension version 2.11.2

[✓] Connected devices (1 available)
    • Android SDK built for x86 • emulator-5554 • android-x86 • Android 8.1.0 (API 27) (emulator)

• No issues found!
```

Other data points:
- I am also encountering this on another unrelated Flutter project.
- @mjohnsullivan was encountering this issue too, on a version of the unit converter project.

**Comment by maryx** [comment]
Thanks for the instructions!

I imported an existing project with the defaults, and it seems like my lib/ directory is not being read, even though it still exists. 
![screenshot from 2018-04-26 17-43-17](https://user-images.githubusercontent.com/3357543/39338771-c3236d04-4979-11e8-919c-68a4fd8f792a.png)

A diff shows that importing the project generated a directory at 	`<project>/android/app/src/main/gen/`. Good news is that yes, the AVD Manager does show up.

I then just tried to replace the string WEB_MODULE with JAVA_MODULE but that didn't change anything.

I noticed that when I made a fresh project from scratch a couple weeks ago, the AVD manager showed up. This was true of https://github.com/flutter/udacity-course/tree/master/course/12_error/solution_12_error when I first created it. However, since re-installing Android Studio, the AVD Manager also doesn't show up for this project either, even though its .iml says JAVA_MODULE.

**Comment by stevemessick** [comment]
It worked for me.
<img width="346" alt="screen shot 2018-04-27 at 5 59 07 am" src="https://user-images.githubusercontent.com/8518285/39363749-3a71953c-49e0-11e8-860e-5b7a1fe3a6b9.png">
As you can see, the Dart sources are processed and there are no errors (after I did `flutter packages get`).

Changing WEB_MODULE to JAVA_MODULE by itself won't do anything. You need to define an Android facet to make the project behave properly in Android Studio. That's what importing does.

It looks like your project is missing all the non-Android directories (ios, third_party, etc) in addition to lib. I have no idea why that would happen. And, your import connected to Android API 26 while mine got API 25. I'm using a clone of the github repos, unmodified. When importing the project I navigated to udacity-course-master/unit_converter/unit_converter and selected that directory. I tried starting the import from different project roots and got the expected results. I also tried using Open instead of Import but nothing I did would produce the project tree you showed. I also had no problem with the solution_12_error project.

I don't have any really good suggestions left. You might try upgrading Android Studio to 3.1.2 and Flutter to M24. Or try on a Mac instead of Linux.

@pq @devoncarew : Any ideas about what happened to import to produce that strange project tree?

**Comment by maryx** [comment]
I just tried again with a fresh git clone of that project, on a Mac. Got the same issue. You're right, the associated ios/third_party/other dirs have also disappeared. I only have the Android 26 and 27 APIs installed on my machine, so it selected 26.

Now going to update Android Studio and Flutter and see if that resolves it.

**Comment by maryx** [comment]
Ah! Resolved.
1. My Android Studio was up to date (3.1.2) but my Flutter was on version 23. I updated it to 24.
2. Previously, every import's Source Files section would look something like this.
<img width="791" alt="screen shot 2018-04-27 at 10 53 06 am" src="https://user-images.githubusercontent.com/3357543/39377492-30be43bc-4a0a-11e8-88d8-6f8843b3b8d2.png">

3. With the latest Flutter, note that Flutter is an option in here, instead of just Java and Android.
![screenshot from 2018-04-27 10-55-31](https://user-images.githubusercontent.com/3357543/39377341-a5b174e2-4a09-11e8-895d-a46af84d0b26.png)

So now I have what you have:
![screenshot from 2018-04-27 10-54-54](https://user-images.githubusercontent.com/3357543/39377393-d3737a1a-4a09-11e8-9270-21a9f655ffee.png)

I think I should be able to delete the /main dir. I note my .iml files have updated too.
Thanks for the suggestion and taking the time to test it out, I'll close this issue!
